typescript/src/link/link.ts

```diff
@@ -175,14 +194,88 @@ export async function parseAndStoreLink<A, B>(
+                            for each acquisition notification from the users mobile apps hitting the endpoints
+                                /apple/linkToSubscriptions
+                                /google/linkToSubscriptions
+
+                            This change is... temporary and a stepping stone to what is going to be version 1.5
+                            and then later on version 2.
+
+                            ### Specifications
+
+                            The trigger for the soft opt in will be the following query parameter:
+                            ```
+                            soft-opt-in-notification-shown=true
+                            ```
+
+                            In this current HTTP request we have a contextual user (corresponding to the userId
+                            which was extracted during authentication) as well as an array of subscriptions
+                            from the request payload. We want to update the dynamo table with soft opt in
+                            information for the subscriptions that have not yet been soft opted in, and we
+                            also post a consent object to the Identity API (once) if at least one of those
+                            subscriptions needed to be soft opted in.
+
+                            ### Implementation details.
+
+                            The type of the payload depends on the platform. It's AppleLinkPayload for iOS
+                            and GoogleLinkPayload for android. Working from the payload here would not be
+                            practical, but since we are only after the subscriptionId, we can read it in both cases
+                            from a UserSubscription.
+
+                            Note that toUserSubscription(userId, payload) return an array of such subscriptions
+
+                            ### Why using a metric driven alerting instead of erroring ?
+
+                            These end points have a very strict contract with the mobile apps about which HTTP
+                            error codes to return in which situation, therefore there should not really be any uncaught
+                            error that filter up to the clients.
+
+                            ### Identity API
+
+                            We are using and have created mp-soft-opt-in-identity-api-key just for this.
+                            We could not use the userAuthenticationToken to post the consent object to Identity
+                            because it doesn't carry the right scopes.
+
+                        */
+
+                        if (softOptInQueryParameterIsPresent(httpRequest)) {
+                            console.log(`softOptInQueryParameterIsPresent`);
+                            const userAuthenticationToken = getAuthToken(httpRequest.headers) as string;
```

Comment:
```suggestion
                            // const userAuthenticationToken = getAuthToken(httpRequest.headers) as string;
```

Maybe comment this out if it's not being used (or remove if it's not going to be bought back later)